lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -89,5 +92,13 @@ def related_replica_count
     def external_name_svc?
       @definition["spec"]["type"] == "ExternalName"
     end
+
+    def requires_publishing?
+      @definition["spec"]["type"] == "LoadBalancer"
+    end
+
+    def published?
+      !@instance_data.dig('status', 'loadBalancer', 'ingress', 'ip').nil?
```

Comment:
I find `present?`  easier to read than  `!X.nil?`

Reply:
Actually had to change this completely; the data structure had a different shape than what I expected. 🎩 FTW!